source/Octopus.Tentacle/Maintenance/WorkspaceCleanerTask.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+
+namespace Octopus.Tentacle.Maintenance
+{
+    public interface IWorkspaceCleanerTask
+    {
+        void Start();
+        void Stop();
+    }
+
+    class WorkspaceCleanerTask : IWorkspaceCleanerTask, IDisposable
+    {
+        readonly WorkspaceCleanerConfiguration configuration;
+        readonly WorkspaceCleaner workspaceCleaner;
+        readonly ISystemLog log;
+
+        readonly CancellationTokenSource cancellationTokenSource = new ();
+        readonly object taskLock = new();
+
+        Task? cleanerTask;
+
+        public WorkspaceCleanerTask(
+            WorkspaceCleanerConfiguration configuration,
+            WorkspaceCleaner workspaceCleaner, 
+            ISystemLog log)
+        {
+            this.configuration = configuration;
+            this.workspaceCleaner = workspaceCleaner;
+            this.log = log;
+        }
+
+        public void Dispose()
+        {
+            Stop();
+
+            cancellationTokenSource.Dispose();
+        }
+
+        public void Start()
+        {
+            lock (taskLock)
+            {
+                cleanerTask = Task.Run(RunTask);
```

Comment:
nit: could this allow many cleaner task to be run if start was called more than once

Reply:
It could in theory. But in practice, this is called from `RunAgentCommand`, and that command would fail much earlier if that is started more than once.

Regardless, I added a fix 🙂